services/blockstorage/commit_block.go

```diff
@@ -65,7 +60,43 @@ func (s *service) commitBlock(ctx context.Context, input *services.CommitBlockIn
 		})
 	}
 
-	logger.Info("committed a block", logfields.BlockHeight(txBlockHeader.BlockHeight()), log.Int("num-transactions", len(input.BlockPair.TransactionsBlock.SignedTransactions)))
+	logger.Info("committed a block", logfields.BlockHeight(proposedBlockHeight), log.Int("num-transactions", len(input.BlockPair.TransactionsBlock.SignedTransactions)))
 
 	return nil, nil
 }
+
+func (s *service) detectForks(proposedBlock *protocol.BlockPairContainer, logger log.Logger) error {
```

Comment:
Consider adding a unit test for this logic as it's self contained

Reply:
it's already covered by component tests. but I've added unit tests now.